### PR TITLE
Update Deploying an application using odo Section and Fixing Broken Links

### LINKED ISSUE #1699: Update Deploying an application using odo section
[kind/Documentation]

A few notes on the [Deploying an application using odo section](https://github.com/openshift/odo#deploying-an-application-using-odo):

It would be helpful to link to the [Minishift GitHub repository](https://github.com/minishift/minishift) or [documentation](https://docs.okd.io/latest/minishift/getting-started/index.html) in step 1:

```
1. Start a local OpenShift development cluster by running minishift.

$ minishift start
```

In `odo` version `v1.0.0-beta1` the command `odo app create` is no longer a valid command, but it is still listed as step 3:

```
3. Create an application. An application in odo is an umbrella under which you add other components.

$ odo app create node-example-app
```

## What did you find missing in the documentation?

It would be nice to include a step 8 to delete the example application using `odo delete` to clean up the `Minishift` project. 

## What is the relevance of it?

These documentation updates will help eliminate confusion for those just getting started with `odo`.

### LINKED ISSUE #1704: Broken Links in README.adoc
The main `README.adoc` for `odo` contains broken links under the following sections: [Installing odo](https://github.com/openshift/odo#installing-odo), [Deploying an application using odo](https://github.com/openshift/odo#deploying-an-application-using-odo), and [Community, discussion, contribution, and support](https://github.com/openshift/odo#community-discussion-contribution-and-support).

This is due to the fact that these links are still pointing to `.md` instead of `.adoc`.

## What did you find missing in the documentation?
Links to appropriate documentation.

## What is the relevance of it?
Makes it difficult for users to find additional information.

### PR DESCRIPTION
**Pull request by danielhelfand**
## What is the purpose of this change? What does it change?
This pull request updates `odo`'s `REAMDME` to fix steps in the [Deploying an application using odo](https://github.com/openshift/odo#deploying-an-application-using-odo) section of the documentation. These updates include the following:

1. Including `Minishift` documentation for Step 1 in case users don't have `Minishift` installed.
2. Removing step 3 `odo app create`, which is no longer used in `v1.0.0-beta1`.
3. Adding a step to delete the component created as part of the tutorial.

This change also addresses broken links in the README as referenced in #1704.

## Was the change discussed in an issue?
fixes #1699
fixes #1704  

## How to test changes?
Go through the new steps listed and verify you are able to create a component, expose its url, access the `nodejs` application, and delete the component.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -106,7 +106,7 @@ https://asciinema.org/a/225717[image:https://asciinema.org/a/225717.svg[asciicas
 After installing `odo`, follow these steps to build, push, and
 deploy a Node.js application. Examples for other supported languages and runtimes can be found link:https://github.com/openshift/odo/blob/master/docs/examples.md[here].
 
-. Start a local OpenShift development cluster by running minishift.
+. Start a local OpenShift development cluster by using `Minishift`. More information on how to install and use Minishift can be found link:https://docs.okd.io/latest/minishift/getting-started/index.html[here], but, if properly set up already, run the following:
```
Comment (mik-dass) [inline README.adoc]: I guess we should remove the `but` from the last from sentence

Comment (surajnarwade) [reply README.adoc]: +1

**Comment by kadel** [comment]
/lgtm
e2e-scenario is failing on login issues again :-(
/retest
```
error: You must be logged in to the server (Unauthorized)
error: You must be logged in to the server (the server has asked for the client to provide credentials)
```

**Comment by kadel** [comment]
/retest
```
[odo]  ✗  Please log in to the cluster
```

**Pull request merged**